lib/ansible/modules/cloud/azure/azure_rm_sqldatabase.py

```diff
@@ -0,0 +1,483 @@
+                self.parameters["create_mode"] = kwargs[key]
+            elif key == "source_database_id":
+                self.parameters["source_database_id"] = kwargs[key]
+            elif key == "source_database_deletion_date":
+                self.parameters["source_database_deletion_date"] = kwargs[key]
+            elif key == "restore_point_in_time":
+                self.parameters["restore_point_in_time"] = kwargs[key]
+            elif key == "recovery_services_recovery_point_resource_id":
+                self.parameters["recovery_services_recovery_point_resource_id"] = kwargs[key]
+            elif key == "edition":
+                self.parameters["edition"] = kwargs[key]
+            elif key == "max_size_bytes":
+                self.parameters["max_size_bytes"] = kwargs[key]
+            elif key == "requested_service_objective_id":
+                self.parameters["requested_service_objective_id"] = kwargs[key]
+            elif key == "requested_service_objective_name":
+                self.parameters["requested_service_objective_name"] = kwargs[key]
+            elif key == "elastic_pool_name":
+                self.parameters["elastic_pool_name"] = kwargs[key]
+            elif key == "read_scale":
+                self.parameters["read_scale"] = kwargs[key]
+            elif key == "sample_name":
+                self.parameters["sample_name"] = kwargs[key]
+            elif key == "zone_redundant":
+                self.parameters["zone_redundant"] = kwargs[key]
+
+        old_response = None
+        results = dict()
+
+        self.mgmt_client = self.get_mgmt_svc_client(SqlManagementClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        try:
+            resource_group = self.get_resource_group(self.resource_group)
+        except CloudError:
+            self.fail('resource group {0} not found'.format(self.resource_group))
+
+        if not ("location" in self.parameters):
+            self.parameters["location"] = resource_group.location
+
+        old_response = self.get_sqldatabase()
+
+        if not old_response:
+            self.log("SQL Database instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Old instance didn't exist")
+            else:
+                self.to_do = Actions.Create
+        else:
+            self.log("SQL Database instance already exists")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                self.log("Need to check if SQL Database instance has to be deleted or may be updated")
+                self.to_do = Actions.Update
+
+        if (self.to_do == Actions.Create) or (self.to_do == Actions.Update):
+            self.log("Need to Create / Update the SQL Database instance")
+
+            if self.check_mode:
+                return self.results
```

Comment:
This would need `results['changed'] = True` in the create scenario. We really need a way to checking if we need to update the database instead of doing the operation and checking if it changed there as well.